NoSuchKeyException when trying to retrieve files and directories using typeFilter FILES_AND_FOLDERS with the AmazonS3Sender

### Current Behavior

```java
Caused by: software.amazon.awssdk.services.s3.model.NoSuchKeyException: (Service: S3, Status Code: 404, Request ID: 1821F0A990A8B1F5, Extended Request ID: dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8) (Service: S3, Status Code: 404, Request ID: 1821F0A990A8B1F5, Extended Request ID: dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8)
	at software.amazon.awssdk.services.s3.model.NoSuchKeyException$BuilderImpl.build(NoSuchKeyException.java:126) ~[s3-2.29.1.jar:?]
	at software.amazon.awssdk.services.s3.model.NoSuchKeyException$BuilderImpl.build(NoSuchKeyException.java:80) ~[s3-2.29.1.jar:?]
	at software.amazon.awssdk.services.s3.internal.handlers.ExceptionTranslationInterceptor.modifyException(ExceptionTranslationInterceptor.java:65) ~[s3-2.29.1.jar:?]
	at software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain.modifyException(ExecutionInterceptorChain.java:181) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.http.pipeline.stages.utils.ExceptionReportingUtils.runModifyException(ExceptionReportingUtils.java:54) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.http.pipeline.stages.utils.ExceptionReportingUtils.reportFailureToInterceptors(ExceptionReportingUtils.java:38) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:39) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.http.pipeline.stages.ExecutionFailureExceptionReportingStage.execute(ExecutionFailureExceptionReportingStage.java:26) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.http.AmazonSyncHttpClient$RequestExecutionBuilderImpl.execute(AmazonSyncHttpClient.java:210) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.invoke(BaseSyncClientHandler.java:103) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.doExecute(BaseSyncClientHandler.java:173) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.lambda$execute$1(BaseSyncClientHandler.java:80) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.measureApiCallSuccess(BaseSyncClientHandler.java:182) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:74) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:45) ~[sdk-core-2.29.1.jar:?]
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:53) ~[aws-core-2.29.1.jar:?]
	at software.amazon.awssdk.services.s3.DefaultS3Client.headObject(DefaultS3Client.java:6575) ~[s3-2.29.1.jar:?]
	at org.frankframework.filesystem.AmazonS3FileSystem.getFileAttributes(AmazonS3FileSystem.java:361) ~[frankframework-aws-9.1.0-SNAPSHOT.jar:9.1.0-SNAPSHOT]
	at org.frankframework.filesystem.AmazonS3FileSystem.updateFileAttributes(AmazonS3FileSystem.java:345) ~[frankframework-aws-9.1.0-SNAPSHOT.jar:9.1.0-SNAPSHOT]
	at org.frankframework.filesystem.AmazonS3FileSystem.getFileSize(AmazonS3FileSystem.java:528) ~[frankframework-aws-9.1.0-SNAPSHOT.jar:9.1.0-SNAPSHOT]
	at org.frankframework.filesystem.AmazonS3FileSystem.getFileSize(AmazonS3FileSystem.java:76) ~[frankframework-aws-9.1.0-SNAPSHOT.jar:9.1.0-SNAPSHOT]
	at org.frankframework.filesystem.FileSystemUtils.getFileInfo(FileSystemUtils.java:314) ~[frankframework-filesystem-9.1.0-SNAPSHOT.jar:9.1.0-SNAPSHOT]
	at org.frankframework.filesystem.FileSystemActor.processListAction(FileSystemActor.java:478) ~[frankframework-filesystem-9.1.0-SNAPSHOT.jar:9.1.0-SNAPSHOT]
	at org.frankframework.filesystem.FileSystemActor.doAction(FileSystemActor.java:402) ~[frankframework-filesystem-9.1.0-SNAPSHOT.jar:9.1.0-SNAPSHOT]
```

### Expected Behavior

-

### Environment Information

 -

### Steps To Reproduce

 -

### Configuration

```xml
<AmazonS3Sender action="list" clientRegion="${client_region}"
    accessKey="${access_key}" secretKey="${secret_key}" bucketName="${bucket_name}"
    serviceEndpoint="${service_endpoint}" typeFilter="FILES_AND_FOLDERS">
</AmazonS3Sender>
```

### Input

_No response_

### What database are you using?

_No response_

### What browsers are you seeing the problem on?

_No response_

### Relevant Log Output

```console

```

### Anything else?

_No response_

--- a/aws/src/main/java/org/frankframework/filesystem/AmazonS3FileSystem.java
+++ b/aws/src/main/java/org/frankframework/filesystem/AmazonS3FileSystem.java
@@ -340,7 +340,7 @@ public class AmazonS3FileSystem extends AbstractFileSystem<S3FileRef> implements
 	 * @throws FileSystemException if it cannot find the resource in S3.
 	 */
 	private void updateFileAttributes(S3FileRef f) throws FileSystemException {
-		if (f.getContentLength() == null) {
+		if (f.getContentLength() == null && f.hasName()) {
 			try {
 				getFileAttributes(f);
 			} catch (AwsServiceException e) {
@@ -525,6 +525,11 @@ public class AmazonS3FileSystem extends AbstractFileSystem<S3FileRef> implements
 
 	@Override
 	public long getFileSize(S3FileRef f) throws FileSystemException {
+		// Probably a folder, which cannot have a size.
+		if (!f.hasName()) {
+			return 0;
+		}
+
 		updateFileAttributes(f);
 		return f.getContentLength();
 	}
@@ -532,6 +537,9 @@ public class AmazonS3FileSystem extends AbstractFileSystem<S3FileRef> implements
 	@Override
 	public Date getModificationTime(S3FileRef f) throws FileSystemException {
 		updateFileAttributes(f);
+		if (f.getLastModified() == null) {
+			return null;
+		}
 		return Date.from(f.getLastModified());
 	}
 

--- a/aws/src/main/java/org/frankframework/filesystem/S3FileRef.java
+++ b/aws/src/main/java/org/frankframework/filesystem/S3FileRef.java
@@ -114,13 +114,17 @@ public class S3FileRef {
 
 	/** Returns either the file or foldername (suffixed with a slash) */
 	public String getName() {
-		if(StringUtils.isNotEmpty(name)) { // File: when not empty, return this immediately
+		if(hasName()) { // File: when not empty, return this immediately
 			return name;
 		}
 
 		// Folder: only take part before last slash
 		String removeEndSlash = StringUtils.chop(folder);
 		return StringUtils.substringAfterLast(removeEndSlash, '/');
+	}
+
+	public boolean hasName() {
+		return StringUtils.isNotEmpty(name);
 	}
 
 	@Nonnull

--- a/aws/src/test/java/org/frankframework/senders/AmazonS3SenderTest.java
+++ b/aws/src/test/java/org/frankframework/senders/AmazonS3SenderTest.java
@@ -9,6 +9,8 @@ import static org.mockito.Mockito.spy;
 import java.io.IOException;
 import java.nio.file.Path;
 
+import org.frankframework.filesystem.TypeFilter;
+
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.io.TempDir;
@@ -121,4 +123,21 @@ public class AmazonS3SenderTest extends WritableFileSystemSenderTest<AmazonS3Sen
 		IOException e = assertThrows(IOException.class, result::preserve); // read binary stream twice
 		assertEquals("Attempted read on closed stream.", e.getMessage());
 	}
+
+	@Test
+	public void fileSystemSenderTestListFolders() throws Exception {
+		fileSystemSender.setAction(FileSystemAction.LIST);
+		fileSystemSender.setTypeFilter(TypeFilter.FOLDERS_ONLY);
+
+		fileSystemSender.configure();
+		fileSystemSender.start();
+
+		createFile("OtherFolder", "OtherFile", "Content of the file!");
+
+		result = fileSystemSender.sendMessageOrThrow(new Message(""), session);
+
+		assertFolderCountEquals(result, 1);
+		assertFileCountEquals(result, 0);
+	}
+
 }

--- a/filesystem/src/test/java/org/frankframework/filesystem/FileSystemSenderTest.java
+++ b/filesystem/src/test/java/org/frankframework/filesystem/FileSystemSenderTest.java
@@ -28,7 +28,7 @@ public abstract class FileSystemSenderTest<FSS extends AbstractFileSystemSender<
 	static final String FOLDER_NAME = "folder";
 	protected FSS fileSystemSender;
 	SenderResult senderResult;
-	Message result;
+	protected Message result;
 
 	public abstract FSS createFileSystemSender();
 

